ext/curl/tests/curl_setopt_CURLOPT_PREREQFUNCTION.phpt

```diff
@@ -0,0 +1,50 @@
+--TEST--
+Curl option CURLOPT_PREREQFUNCTION
+--EXTENSIONS--
+curl
+--SKIPIF--
+<?php
+$curl_version = curl_version();
+if ($curl_version['version_number'] < 0x075000) die("skip: test works only with curl >= 7.80.0");
+?>
+--FILE--
+<?php
+include 'server.inc';
+
+$ch = curl_init();
+
+$host = curl_cli_server_start();
+$url = "{$host}/get.inc?test=";
+curl_setopt($ch, CURLOPT_URL, $url);
```

Comment:
do you think "enforcing" ip address kind with `CURLOPT_IPRESOLVE` would help ?

Reply:
Thank you so much! There was indeed a test failure because the test server had Curl server running on IPv6 loopback address. I tried "forcing" with IPv4, but had the tests failing for some reason. I ended up adding `filter_var` calls to check that the values passed are IP addresses, with tests passing now.

Reply:
alright